[JIT] Remove dead store in quantization_patterns.h

SplitInfinity opened the pull request:
Test Plan: Continuous integration.

Differential Revision: D22259209

jerryzh168 commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -136,7 +136,6 @@ QuantFusionInfo getBinaryOpScalarFusionInfo(
   std::string op_pattern =
       getInputTensorQParamOpPattern(op_name, extra_op_args);
 
-  const auto& extra_op_arg_list = getExtraArgList(extra_op_args);
```
isn't this used in the next line?

SplitInfinity replied:
I requested review before checking tests and then immediately unrequested, sorry about that. Let me figure out what's going on. This dead store was flagged by an internal tool.

jerryzh168 replied:
maybe it's related to `const auto&`, you can try changing to `auto`

SplitInfinity commented:
I requested review before checking tests and then immediately unrequested, sorry about that. Let me figure out what's going on. This dead store was flagged by an internal tool.

Commit:
[JIT] Remove dead store in quantization_patterns.h (#40623)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/40623

Test Plan: Continuous integration.

Differential Revision: D22259209

fbshipit-source-id: 2cf7869a9bd7a8560acf0aeca8f487597e112c09

jerryzh168 reviewed (approved):
I see, thanks!
